### PR TITLE
Add extensions for `Timestamp` and `Duration`

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  * updates dependencies, addressing warnings and fixing changed package names.
  * Adds Kotlin extensions for `Timestamp` and `Duration`.
  * Adds the extension `java.time.Instant.toTimestamp()`.
  * Extends `Temporal` with the comparison methods  accepting `Timestamp` and `Instant`. 
     This will allow to avoid creation of corresponding `Temporal<Timestamp>` and `Temporal<Instant>` instances just for the comparison.

### CODE REVIEW

File: time/src/main/kotlin/io/spine/time/DurationExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+/**
+ * Returns `true` if the passed value is zero, `false` otherwise.
+ *
+ * @see Durations2.ZERO
+ */
+public fun Duration.isZero(): Boolean = isZero(this)
+
+/**
+ * Convert Duration to string format.
```
Comment (armiol) [inline time/src/main/kotlin/io/spine/time/DurationExtensions.kt]: `Converts`.

It's here and below.

File: time/src/main/kotlin/io/spine/time/DurationExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+ *
+ * The string format will contains 3, 6, or 9 fractional digits depending on the precision required
```
Comment (armiol) [inline time/src/main/kotlin/io/spine/time/DurationExtensions.kt]: `will contain`.

File: time/src/main/kotlin/io/spine/time/DurationExtensions.kt [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+ * to represent the exact `Duration` value. For example: "1s", "1.010s", "1.000000100s", "-3.100s".
+ * The range that can be represented by `Duration` is from -315,576,000,000 to +315,576,000,000
+ * inclusive (in seconds).
+ *
+ * @return The string representation of the given duration.
```
Comment (armiol) [inline time/src/main/kotlin/io/spine/time/DurationExtensions.kt:106]: Do we put periods at the end of `@return` and `@throws` in Kotlin? If not, please remove.

File: time/src/main/kotlin/io/spine/time/TimestampExtensions.kt [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+ *
+ * The [nanos][Timestamp.getNanos] value must be in the range `[0, +999,999,999]`.
+ *
+ * **Note:** Negative second values with fractional seconds must still have non-negative
+ * nanos values that count forward in time.
+ */
+public fun Timestamp.isValid(): Boolean = isValid(this)
+
+/**
+ * Convert this timestamp to RFC 3339 date string format.
```
Comment (armiol) [inline time/src/main/kotlin/io/spine/time/TimestampExtensions.kt]: This file also should have `Converts`.

**Review by armiol — APPROVED** [review]
@alexander-yevsyukov LGTM with some comments to address before merging.

File: time/src/main/kotlin/io/spine/time/DurationExtensions.kt (hunk 3 above)
Comment (alexander-yevsyukov) [reply time/src/main/kotlin/io/spine/time/DurationExtensions.kt:106]: From what I see in `kotlin.collections` classes, they do put periods in the end of sentences, which I find a more correct from the language point of view.

**Pull request merged**